solidity/contracts/bridge/Bridge.sol

```diff
@@ -773,4 +796,191 @@ contract Bridge is Ownable {
     //      depositors, next to sweep, to prove their swept balances on Ethereum
     //      selectively, based on deposits they have earlier received.
     //      (UPDATE PR #90: Is it still the case since amounts are inferred?)
+
+    // TODO: Documentation.
+    function requestRedemption(RedemptionRequest calldata request) external {
+        require(request.redeemer == msg.sender, "Redeemer is not the caller");
+
+        // TODO: Validate wallet choice, specifically whether it contains a
+        //       sufficient BTC balance.
+
+        uint256 redemptionRequestKey =
+            uint256(
+                keccak256(
+                    abi.encodePacked(
+                        request.redeemer,
+                        request.redeemerOutputHash,
+                        redeemer.walletPubKeyHash,
+                        redeemer.amount
+                    )
+                )
+            );
+
+        require(
+            redemptionRequests[redemptionRequestKey] == uint32(0),
+            "Redemption request already registered"
+        );
+        /* solhint-disable-next-line not-rely-on-time */
+        redemptionRequests[redemptionRequestKey] = uint32(block.timestamp);
+
+        emit RedemptionRequested(
+            request.redeemer,
+            request.redeemerOutputHash,
+            redeemer.walletPubKeyHash,
+            redeemer.amount
+        );
+
+        bank.transferBalanceFrom(
+            request.redeemer,
+            address(this),
+            request.amount
+        );
```

Comment:
I think it makes sense because the wallet needs to "reserve" the given Bank balance. For this PR, I would assume `Bridge` has this amount approved and I'd look at the flow separately.

Reply:
Yep, that's right.